src/helpers/load.ts

```diff
@@ -48,11 +48,22 @@ function handleExecutionError(reason: IError) {
     }
 }
 
-function decorateMetrics(visObj: VisualizationObject.IVisualizationObject):
-    VisualizationObject.IVisualizationObject {
+function calculateLocalIdentifier(
+    item: VisualizationObject.IMeasure,
+    index: number
+): string {
+    // reuse identifier provided by AD
+    const localIdentifier = get(item, ['measure', 'localIdentifier'], '');
+    // fallback when vis. object provided from KD without localIdentifiers
+    return localIdentifier.length ? localIdentifier : `m${index + 1}`;
+}
+
+function decorateMetrics(
+    visObj: VisualizationObject.IVisualizationObject,
+): VisualizationObject.IVisualizationObject {
     const updatedVisObj = cloneDeep(visObj);
     updatedVisObj.content.buckets.measures = updatedVisObj.content.buckets.measures.map((measure, index) => {
-        measure.measure.generatedId = `m${index + 1}`;
+        measure.measure.localIdentifier = calculateLocalIdentifier(measure, index);
```

Comment:
just a suggestion, what about `getLocalIdentifier`(calculate implies to me some "sophisticated" calculation :-) )